### PR TITLE
Migrate harbor storage cluster

### PR DESCRIPTION
**Pull request by soulseen**

### CODE REVIEW

File: pkg/cluster/storage/minio.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+var (
+	HarborClusterMinIOGVK = schema.GroupVersionKind{
+		Group:   minio.SchemeGroupVersion.Group,
+		Version: minio.SchemeGroupVersion.Version,
+		Kind:    minio.MinIOCRDResourceKind,
+	}
+)
+
+// TODO soulsee
+func NewMinIOController() lcm.Controller {
```
Comment (steven-zou) [inline pkg/cluster/storage/minio.go:63]: I think we can start to update this new func

File: pkg/cluster/storage/minio.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+			LastTransitionTime: metav1.Now(),
+			Reason:             "",
+			Message:            "",
+		},
+		Properties: *properties,
+	}
+}
+
+// TODO Deprecated
+func (m *MinIOController) ScaleUp(newReplicas uint64) (*lcm.CRStatus, error) {
```
Comment (steven-zou) [inline pkg/cluster/storage/minio.go:239]: Remove?

**Review by steven-zou — COMMENTED** [review]
commented

File: pkg/cluster/storage/provision.go [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (m *MinIOController) ProvisionMinIOProperties(minioInstamnce *minio.Tenant) (*lcm.CRStatus, error) {
+	properties := &lcm.Properties{}
+	data,err := m.getMinIOProperties(minioInstamnce)
+	if err != nil {
+		return minioNotReadyStatus(getMinIOProperties, err.Error()), err
+	}
+	properties.Add(lcm.InClusterSecretForStorage, data)
```
Comment (wangcanfengxs) [inline pkg/cluster/storage/provision.go:29]: I think incluster and external mode will return the same property key, so using `storage` as key directly?

**Review by wangcanfengxs — COMMENTED** [review]
comments left

File: pkg/cluster/storage/provision.go (hunk 3 above)
Comment (soulseen) [reply pkg/cluster/storage/provision.go:29]: right, this name may be confused, I will update it, in actually, LCM-controller not have any logic about external storage.

File: pkg/cluster/storage/delete.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package storage
+
+import (
+	goharborv1 "github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+	"github.com/goharbor/harbor-operator/pkg/lcm"
+)
+
+func (m *MinIOController)  Delete(harborcluster *goharborv1.HarborCluster) (*lcm.CRStatus, error) {
```
Comment (chlins) [inline pkg/cluster/storage/delete.go:8]: go fmt?

**Review by wangcanfengxs — APPROVED** [review]

**Review by Ghostbaby — APPROVED** [review]
lgtm

**Pull request merged**